src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -4676,12 +4677,15 @@ public class ShowProteinAction extends SimpleViewAction<ChromatogramForm>
         @Override
         public ModelAndView getView(final ChromatogramForm form, BindException errors)
         {
+            String viewBy = getViewContext().getActionURL().getParameter("viewBy");
```

Comment:
Suggest adding this to the form object instead of pulling it separately from the URL. May want to introduce a ChromatogramForm subclass